src/components/ErrorBoundry/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { PureComponent } from 'react';
+import { ErrorLayout } from 'components/ErrorLayout';
+
+interface ErrorBoundaryProps {
+    children: JSX.Element;
+}
+
+interface ErrorBoundaryState {
+    hasError: boolean;
+    err: Error | null;
+}
+
+export class ErrorBoundary extends PureComponent<ErrorBoundaryProps, ErrorBoundaryState> {
+    constructor(props: ErrorBoundaryProps) {
+        super(props);
+        this.state = { hasError: false, err: null };
+    }
+
+    static getDerivedStateFromError(err: Error): ErrorBoundaryState {
+        return { hasError: true, err };
+    }
+
+    override componentDidCatch(err: Error, errorInfo: React.ErrorInfo): void {
```

Comment:
Нет необходимости в `override`. Единственный кейс для его использования - в базовом классе, чтобы проверить, что наследники реализуют указанный метод

Reply:
Без этого модификатора ts выдает ошибку: This member must have an 'override' modifier because it overrides a member in the base class 'PureComponent<ErrorBoundaryProps, ErrorBoundaryState, any>'.

Reply:
◕_◕